spec/feature/remember_me_spec.rb

```diff
@@ -0,0 +1,95 @@
+RSpec.feature "Remember Me" do
+  include ActiveJob::TestHelper
+  include ActiveSupport::Testing::TimeHelpers
+
+  before { allow(Rails.configuration).to receive(:feature_flag_mfa).and_return(true) }
+  before { allow(Rails.configuration).to receive(:feature_flag_bypass_mfa).and_return(bypass_mfa_enabled) }
+
+  let(:bypass_mfa_enabled) { true }
+
+  let!(:user) { FactoryBot.create(:user, :confirmed) }
+
+  context "'remember me' is disabled" do
+    let(:bypass_mfa_enabled) { false }
+
+    it "doesn't give the option to 'remember me'" do
+      enter_email_address_and_password
+
+      expect(page).to_not have_text(Rails::Html::FullSanitizer.new.sanitize(I18n.t("mfa.phone.code.fields.remember_me.label")))
+    end
+  end
+
+  context "the user checks 'remember me'" do
+    before { log_in_and_remember_me }
+
+    it "shows the event on the security page" do
+      visit_security_page
+
+      expect(page).to have_text(I18n.t("account.security.security_codes.code_description.present"))
+    end
+
+    context "the user returns 29 days later" do
+      before do
+        travel(MultiFactorAuth::BYPASS_TOKEN_EXPIRATION_AGE - 1.day)
+        enter_email_address_and_password
+      end
+
+      it "skips MFA" do
+        expect(page).to have_text(I18n.t("account.your_account.heading"))
+      end
+    end
+
+    context "the user returns 31 days later" do
+      before do
+        travel(MultiFactorAuth::BYPASS_TOKEN_EXPIRATION_AGE + 1.day)
+        enter_email_address_and_password
+      end
+
+      it "does not skip MFA" do
+        expect(page).to have_text(I18n.t("mfa.phone.code.sign_in_heading"))
+      end
+    end
+
+    context "with multiple users on the same machine" do
+      let(:user2) { FactoryBot.create(:user, email: "other-user@example.com") }
+
+      before do
+        travel(Devise.timeout_in + 1.second)
+        log_in_and_remember_me(the_user: user2)
+        travel(Devise.timeout_in + 1.second)
+      end
+
+      it "remembers the first user" do
+        enter_email_address_and_password(the_user: user)
+        expect(page).to have_text(I18n.t("account.your_account.heading"))
+      end
+
+      it "remembers the second user" do
```

Comment:
Took me a moment to grok what's going on with this test.
I iniitally thought it was saying the remember me was pegged to a browser and not an account before I went back over that additional before loop.

Do you think we could rearrage that?
Perhaps we could move lines 56 insie the second test?

I wonder if we could demonstrate that:
- the initial user can still log in fine (L61-L65)
- a new user goes to log in, and _does_ get prompted for MFA
- the user then logs out / returns and they are not prompted for MFA?

We're trying to demostrate they're account specific right?

Reply:
I'll combine these into a single test

Reply:
Ace thanks, the "it" descriptions are great improvements too.
I find that much easier to grok, thanks